### DISCUSSION TITLE
How to create multi-page course pages

### DISCUSSION THREAD

**Discussion by cagix** [LiaScript/LiaScript#193] [Q&A] [answered]
Is there a way to create multi-page course pages with LiaScript? 

So far I have only found examples of how to distribute individual Markdown files with LiaScript. However, my courses usually consist of a kind of overview page (Readme.md) containing links to the individual lessons (also Markdown files somewhere in a folder structure in the same repo). When I render this Readme.md via LiaScript, all the links are present, only the referenced markdown files are not rendered, but displayed in raw format. 

(Of course, I could put every single lesson into ILIAS as a LiaScript course for the students, but that's quite some effort and it's not really nice on the students' mobile phones because every lesson gets then its own “app”.)

I have tried to concatenate all Markdown files as a single large file. But this results in a quite large and virtually unusable menu in the sidebar. In addition, I then have to adapt all local links in the result file into internal links to headings, which is also not a trivial task (due to the folder structure).

Do you have any ideas on how I could implement this? I really like the realisation of the quizzes :)

**Comment by SebastianZug** [comment]
Hi cagix,
what is your actual distribution path? You write the file, generate a SCORM file (or similar) and integrate its content in ILIAS?
Your material is not hosted externally on Github?
Sebastian

**Reply by cagix** [reply]
The course material would be organised in a public repo like https://github.com/Compiler-CampusMinden/CB-Vorlesung-Bachelor/. There is a [readme.md](https://github.com/Compiler-CampusMinden/CB-Vorlesung-Bachelor/blob/master/readme.md) containing the syllabus and roadmap. Each class consists of several lessons, which are individual Markdown files in the same repo, linked from the roadmap.

I tried putting a link like https://liascript.github.io/course/?https://github.com/Compiler-CampusMinden/CB-Vorlesung-Bachelor/blob/master/readme.md in ILIAS. This will render the readme and provide all the links in the roadmap, but when accessing the links to the individual lessons, I end up with raw markdown ...

Edit: In the example above actually no local links are being rendered, probably because we use reference links (https://pandoc.org/MANUAL.html#extension-shortcut_reference_links).

**Reply by SebastianZug** [reply]
@andre-dietrich built a really cool makefile (based on liaex), that transforms a [collection of course parameter](https://github.com/TUBAF-IfI-LiaScript/TUBAF-IfI-LiaScript.github.io) into a website. The result looks like https://tubaf-ifi-liascript.github.io/softwareentwicklung.html.

Probably you can integrate such site as an iframe in ILIAS? Just an idea ...

**Reply by cagix** [reply]
Thanks for this idea, I'll have a look.

Iframes are as of now a "no go", as our IT department deemed this "unsafe" and disabled it. So this is a route I cannot take (anymore).

**Reply by cagix** [reply]
hmmm, without fully understanding @andre-dietrich's script fu, this could be a way 👍 i'd need to translate all local/relative links in my readme to liascriptified urls and publish the resulting readme to ilias. 

however, students will still end up with 20..50 individual lessons/apps (depending on the module/course) on their phones ...

**Comment by andre-dietrich** [comment, marked as answer]
Hi @cagix ... could you try the following:

``` markdown
<!--
@open
<script modify="false" run-once>
const origin = window.location.origin;
const pathname = window.location.pathname;
`LIASCRIPT: [@0](${origin + pathname}?@1)`
</script>
@end
-->

# Course Main Title

This is a link to the first lesson of the course:
@[open(Title of the course)](course/lesson_1.md)
```

This will construct a new LiaScript link and put it out as a result. In my case it looks like this:

![image](https://github.com/user-attachments/assets/ddd36f11-7072-4171-8ac1-783428f46435)

The `@open` macro is defined in the main-comment. It will inject a script to which the title is passed as the first argument and the relative URL is passed as the second. Since the LiaScript interpreter is used either from our website or from within an IMS/SCORM package it needs to be loaded and the Markdown-file is passed as the URL parameter. Such a URL is constructed within the script.

You can replace the `LIASCRIPT: ...` result string also by `HTML: <a href="..."` if you want to change the target for example, so that the course gets loaded in the current file.

---

However, quizzes cannot be stored this way in a LMS if SCORM is used and the course is sepparated into multiple files. The quizzes need to be known before the course is loaded and the main README is analyzed and an index on all quizzes and surveys defined here is stored in the LMS.

---

I hope this makes sense ;-)

**Reply by cagix** [reply]
amazing! thx a lot for this idea! i'll give this a try tomorrow.

will this also work with images in the referenced file or is just plain markdown supported?

**Reply by cagix** [reply]
@andre-dietrich thx 🙏 that works great! even images and code blocks in the referenced markdown file are rendered correctly :)

now i just need to figure out how to _not_ split the content at the headers when in textbook mode, but display it like a ‘normal’ website. but that's probably just a matter of configuring it correctly, i suppose.

**Reply by andre-dietrich** [reply]
Hi, there are multiple options for accomplishing this.

``` markdown
# Main Title

SubTitle
========

SubSubTitle
===========
```

Or, what I think is better for accessibility reasons, you can put this into section and article tags

``` markdown
# Main Title

<section>

## SubTitle

...

</section>

--------------------------------------------------------

<article>

## SubTitle

...

### SubSubTitle

</article>
```

**Reply by cagix** [reply]
Ah, I had read about this and then somehow completely forgotten about it. Thank you so much for this reminder!
